### PR TITLE
Prettify changelog for upcoming release

### PR DESCRIPTION
**Pull request by peterjunpark**
This PR makes CHANGELOG.md's formatting more consistent with other ROCm components ahead of 6.1.1 release.
- Reordered some headings.
- Made headings more consistent with other component docs.
- Added some periods.

### CODE REVIEW

**Review by randyh62 — APPROVED** [review]
Optimizations is just another word for Changes...

**Review by marifamd — CHANGES_REQUESTED** [review]
It turns out our changelog is too verbose. We talked with Roopa and the docs team and will shorten the changelog in a future release. As for the format, we follow the MLSE changelog where we have:

Each release will be in its own section (## level).

Each note will be in a subsection (### level). Please follow subsection-ordering. Subsections are:
- Changes - Behavior or features that change in functionality
- Additions - Features added to the software package
- Optimizations - Place performance optimization items in this section.
- Deprecations - Features that still work in the software package but will no longer be supported in the future.
- Removals - Announcements for deprecated features are removed from the software and no longer supported.
- Fixes - An issue known to the public whose functionality is fixed.
- Known issues - items in this list will roll over in the changelog for each release until they are announced in the “Fixed” subsection. There is an intention to have a issue tracking system available for use in the future. GitHub Issues is the logical choice but approvals are needed.
- Security - Security related announcements that will be tracked as a known issue. These are highlighted separately due to the risks involved.

**Comment by peterjunpark** [comment]
Hi @marifamd, I'm aligned with what you're saying and I'm aware of the agreed-upon template.

>It turns out our changelog is too verbose. We talked with Roopa and the docs team and will shorten the changelog in a future release. As for the format, we follow the MLSE changelog where we have:

The docs team didn't want to include all of your changelog in the consolidated changelog on the ROCm docs portal because it's too long. The solution for 6.1.1 is to add an abbreviated version of your changelog and link to this repo where users can find more details.

Normally, the docs team cleans up each ROCm component's changelog before including it in the ROCm consolidated version. Since we cannot do that for this project and we're linking to this repo, I'd like to make it look consistent with other ROCm projects.

This PR only aligns your changelog's formatting with most of the other ROCm components.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
 - **Added process isolation and clean shader APIs and CLI commands**  
+
 Added APIs CLI and APIs to address LeftoverLocals security issues. Allowing clearing the sram data and setting process isolation on a per GPU basis. New APIs:
   - `amdsmi_get_gpu_process_isolation()`
   - `amdsmi_set_gpu_process_isolation()`
   - `amdsmi_set_gpu_clear_sram_data()`
 
-- **Added `MIN_POWER` to output of `amd-smi static --limit`**  
+- **Added `MIN_POWER` to output of `amd-smi static --limit`**.
 This change was to help users to identify what range they can change the power cap of the GPU to. We added this to simplify why a device supports (or does not support) power capping (also known as overdrive). See `amd-smi set -g all --power-cap <value in W>` or `amd-smi reset -g all --power-cap`.
```
Comment (Rmalavally) [inline CHANGELOG.md]: Looks good, Peter. Minor tweak,

Recommended:

This change helps users identify the range to which they can change the power cap of the GPU. The change is added to simplify why a device supports (or does not support) power capping (also known as overdrive). See `amd-smi set -g all --power-cap <value in W>` or `amd-smi reset -g all --power-cap.`

**Review by Rmalavally — APPROVED** [review]
Looks good, Peter.

**Comment by dmitrii-galantsev** [comment]
perfect, I will merge internally and it will get pushed to github automagically. Thanks!
(please do not do anything else in this PR)

**Pull request merged**